open-metadata-implementation/access-services/subject-area/subject-area-client/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/client/SubjectAreaClient.java

```diff
@@ -9,65 +9,190 @@
 
 import java.util.List;
 
+/**
+ * @param <T> object type for supplied and return.
+ * Interface describing common methods to client working with Subject area resources.
+ */
 public interface SubjectAreaClient<T> {
 
     FindRequest EMPTY_FIND_REQUEST = new FindRequest();
 
+    /**
+     * Get object by guid
+     *
+     * @param userId unique identifier for requesting user, under which the request is performed.
+     * @param guid   unique identifier of the object to which the found objects should relate.
+     * @return found objects of the T type.
+     *
+     * @throws PropertyServerException    something went wrong with the REST call stack.
+     * @throws UserNotAuthorizedException the requesting user is not authorized to issue this request.
+     * @throws InvalidParameterException  one of the parameters is null or invalid.
+     */
     T getByGUID(String userId, String guid) throws InvalidParameterException,
                                                    PropertyServerException,
                                                    UserNotAuthorizedException;
 
+    /**
+     * Create a object. To create, you must pass the created object and specify a unique user identifier.
+     *
+     * @param userId   unique identifier for requesting user, under which the request is performed.
+     * @param supplied object to create.
+     * @return created object.
+     *
+     * @throws PropertyServerException    something went wrong with the REST call stack.
+     * @throws UserNotAuthorizedException the requesting user is not authorized to issue this request.
+     * @throws InvalidParameterException  one of the parameters is null or invalid.
+     */
     T create(String userId, T supplied) throws InvalidParameterException,
                                                PropertyServerException,
                                                UserNotAuthorizedException;
 
+    /**
+     * Request to search all objects of the current type T.
```

Comment:
I think we should have paging on this on call. Or at least a warning for now - getting all Terms when there are 100s of thousands will lill the system.

Reply:
The method is called find but we talk about search we should be consistent

Reply:
Okay.